Cython/Build/Dependencies.py

```diff
@@ -1212,14 +1214,9 @@ def cythonize_one(pyx_file, c_file, fingerprint, cache, quiet, options=None,
     from ..Compiler.Main import compile_single, default_options
     from ..Compiler.Errors import CompileError, PyrexError
 
-    if cache and fingerprint:
-        cached = cache.lookup_cache(c_file, fingerprint)
-        if cached:
-            if not quiet:
-                print("%sFound compiled %s in cache" % (progress, pyx_file))
-            cache.load_from_cache(c_file, cached)
-            return
-    if not quiet:
+    if cache and fingerprint and cache.lookup_cache(c_file, fingerprint) and not quiet:
+        print("%sFound compiled %s in cache" % (progress, pyx_file))
+    elif not quiet:
         print("%sCythonizing %s" % (progress, Utils.decode_filename(pyx_file)))
```

Comment:
`quiet` is the main thing to test here, so let's move it first.
```suggestion
    if not quiet:
        if cache and fingerprint and cache.lookup_cache(c_file, fingerprint):
            print(f"{progress}Found compiled {pyx_file} in cache")
        else:
            print(f"{progress}Cythonizing {Utils.decode_filename(pyx_file)}")
```